Assets/Scripts/Models/Utility.cs

```diff
@@ -0,0 +1,802 @@
+    /// </summary>
+    private Parameter utilParameters;
+
+    private List<Job> jobs;
+
+    private List<Job> pausedJobs;
+
+    // This is the generic type of object this is, allowing things to interact with it based on it's generic type
+    private HashSet<string> typeTags;
+
+    private string name = null;
+
+    private string description = string.Empty;
+
+    private Func<Tile, bool> funcPositionValidation;
+
+    private HashSet<TileType> tileTypeBuildPermissions;
+
+    private bool isOperating;
+
+    /// TODO: Implement object rotation
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Utility"/> class.
+    /// </summary>
+    public Utility()
+    {
+        Tint = new Color(1f, 1f, 1f, .25f);
+        JobSpotOffset = Vector2.zero;
+        EventActions = new EventActions();
+
+        contextMenuLuaActions = new List<ContextMenuLuaAction>();
+        utilParameters = new Parameter();
+        jobs = new List<Job>();
+        typeTags = new HashSet<string>();
+        funcPositionValidation = DefaultIsValidPosition;
+        tileTypeBuildPermissions = new HashSet<TileType>();
+    }
+
+    // Copy Constructor -- don't call this directly, unless we never
+    // do ANY sub-classing. Instead use Clone(), which is more virtual.
+    private Utility(Utility other)
+    {
+        ObjectType = other.ObjectType;
+        Name = other.Name;
+        typeTags = new HashSet<string>(other.typeTags);
+        description = other.description;
+        Tint = other.Tint;
+        LinksToNeighbour = other.LinksToNeighbour;
+
+        JobSpotOffset = other.JobSpotOffset;
+
+        utilParameters = new Parameter(other.utilParameters);
+        jobs = new List<Job>();
+        pausedJobs = new List<Job>();
+
+        if (other.EventActions != null)
+        {
+            EventActions = other.EventActions.Clone();
+        }
+
+        if (other.contextMenuLuaActions != null)
+        {
+            contextMenuLuaActions = new List<ContextMenuLuaAction>(other.contextMenuLuaActions);
+        }
+
+        getSpriteNameAction = other.getSpriteNameAction;
+
+        if (other.funcPositionValidation != null)
+        {
+            funcPositionValidation = (Func<Tile, bool>)other.funcPositionValidation.Clone();
+        }
+
+        tileTypeBuildPermissions = other.tileTypeBuildPermissions;
+
+        LocalizationCode = other.LocalizationCode;
+        UnlocalizedDescription = other.UnlocalizedDescription;
+    }
+
+    /// <summary>
+    /// This event will trigger when the utility has been changed.
+    /// This is means that any change (parameters, job state etc) to the utility will trigger this.
+    /// </summary>
+    public event Action<Utility> Changed;
+
+    /// <summary>
+    /// This event will trigger when the utility has been removed.
+    /// </summary>
+    public event Action<Utility> Removed;
+
+    /// <summary>
+    /// Gets the width of the utility.
+    /// </summary>
+    public int Width 
```

Comment:
Unless there's some way around it that I'm unaware of, these have to be properties due to the interface, which I don't think can be const.